### PR TITLE
artifact_repositories

### PR DESCRIPTION
**Pull request by omsmith**
adds an artifact_repositories input and related secrets allows a user to configure a set of repos to download prior to planning

### CODE REVIEW

File: actions/plan/artifact-repositories/download-one.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+check_slug() {
+	local slug="${1}"; shift
+
+	if [[ ! "${slug}" =~ ^[a-zA-Z0-9][a-zA-Z0-9-]+/[a-zA-Z0-9_.-]+$ ]]; then
```
Comment (cbortos-d2l) [inline actions/plan/artifact-repositories/download-one.sh:8]: Does this regex match what github is using to validate repo names?

File: actions/plan/artifact-repositories/download-many.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+export D2L_ARTIFACT_REPOSITORIES_TMP_DIR=$(mktemp -d)
+trap onexit EXIT
+onexit() {
+	set +u
+
+	rm -r "${D2L_ARTIFACT_REPOSITORIES_TMP_DIR}" 2> /dev/null || true
+}
+
+errors=$(echo "${CONFIG}" \
+	| yq -o=json . - \
```
Comment (cbortos-d2l) [inline actions/plan/artifact-repositories/download-many.sh:14]: Do we want to support YAML here, or does `yq` help for json values too?

Comment (omsmith) [reply actions/plan/artifact-repositories/download-many.sh:14]: I was thinking of just supporting it, yeah. I could see not for consistency sake

File: actions/plan/artifact-repositories/download-one.sh (hunk 1 above)
Comment (omsmith) [reply actions/plan/artifact-repositories/download-one.sh:8]: org/user names, not precisely - it's more restrictive
repo names, should be about right, yeah

The ref names in check_ref, definitely not, but didn't want to think about it too hard, can be relaxed in the future if needed

**Review by cbortos-d2l — APPROVED** [review]

**Pull request merged**